README.md

```diff
@@ -1,89 +1,218 @@
 
-## Mini Tree Generator
+While both `LocalSpawner` and `CloudSpawner` manage Docker containers, they differ in operational context. `LocalSpawner` handles individual containers directly, providing simplicity but limited scalability. In contrast, `CloudSpawner` employs Docker Swarm to manage containers as services, thereby offering enhanced scalability and resilience, crucial for larger workloads.
 
-- Build the Image
+The `ExecutorSpawner` is a trait with minimal requirements, specifically the Rust trait methods `spawn_executor` and `terminate_executor`. You can create your own spawner and use it with the Orchestrator.
+
+## Orchestrating on Swarm
+
+Docker Swarm transforms multiple Docker hosts into a single virtual host, providing crucial capabilities for high availability and scalability in distributed systems like Summa Aggregation.
+
+For more details about Docker Swarm mode, refer to the [official documentation](https://docs.docker.com/engine/swarm/).
+
+### Preparing Docker Swarm Mode
+
+You can initialize your Docker environment in Swarm mode, which is essential for managing a cluster of Docker nodes as a single virtual system.
+
+1. **Activate Swarm Mode on the Main Machine**:
   
-  To build the image, run the following command:
-  ```
-  docker build . -t summa-aggregation/mini-tree
-  ```
+    Run the following command to initialize Swarm mode:
 
-- Run the `Mini Tree Generator Container`
+    ```bash
+    Main $ docker swarm init
+    ```
 
-  Use the command below to start the Mini Tree Generator container:
+      This command will output information about the Swarm, including a join token.
 
-  ```
+2. **Join Worker Nodes to the Swarm**:
+
+      Use the join token provided by the main machine to add worker nodes to the swarm. On each worker node, run like:
+
+      ```bash
+      Worker_1 $ docker swarm join --token <YOUR_JOIN_TOKEN> <MAIN_MACHINE_IP>:2377
+      ```
+
+      Replace `<YOUR_JOIN_TOKEN>` with the actual token and `<MAIN_MACHINE_IP>` with the IP address of your main machine.
+
+3. **Verify Node Status**:
+  
+      To confirm that the nodes are successfully joined to the swarm, check the node status on the main machine:
+
+      ```bash
+      Main $ docker node ls
+      ```
+
+      You should see a list of all nodes in the swarm, including their status, roles, and other details like this:
+
+      ```bash
+      ID                            HOSTNAME   STATUS    AVAILABILITY   MANAGER STATUS   ENGINE VERSION
+      kby50cicvqd5d95o9pgt4puo9 *   main       Ready     Active         Leader           20.10.12
+      2adikgxr2l1zp9oqo4kowvw7n     worker_1   Ready     Active                          20.10.12
+      dz2z2v7o06h6gazmjlspyr5c8     worker_2   Ready     Active                          20.10.12
+      ````
+
+      You are ready to spawn more workers!
+
+### Spawning More Workers
+
```

Comment:
In this context, 'Worker Node' refers to a machine, while 'Worker' denotes a container.

To avoid confusion, I would rename 'Worker Node' to 'Node'.